src/base/bittorrent/session.cpp

```diff
@@ -2753,6 +2761,59 @@ QStringList Session::bannedIPs() const
     return m_bannedIPs;
 }
 
+#if defined(Q_OS_WIN)
+OSMemoryPriority Session::getOSMemoryPriority() const
+{
+    return m_OSMemoryPriority;
+}
+
+void Session::setOSMemoryPriority(const OSMemoryPriority priority)
+{
+    if (m_OSMemoryPriority == priority)
+        return;
+
+    m_OSMemoryPriority = priority;
+    configureDeferred();
+}
+
+void Session::applyOSMemoryPriority() const
+{
+    using SETPROCESSINFORMATION = BOOL (WINAPI *)(HANDLE, PROCESS_INFORMATION_CLASS, LPVOID, DWORD);
+    const auto setProcessInformation = Utils::Misc::loadWinAPI<SETPROCESSINFORMATION>("Kernel32.dll", "SetProcessInformation");
+    if (!setProcessInformation)  // only available on Windows >= 8
+        return;
+
+#if (_WIN32_WINNT < _WIN32_WINNT_WIN8)
+    // this dummy struct is required to compile successfully when targeting older Windows version
+    struct MEMORY_PRIORITY_INFORMATION
+    {
+        ULONG MemoryPriority;
+    };
+#endif
+
+    MEMORY_PRIORITY_INFORMATION prioInfo {};
+    switch (getOSMemoryPriority()) {
+    case OSMemoryPriority::Normal:
+    default:
+        prioInfo.MemoryPriority = MEMORY_PRIORITY_NORMAL;
```

Comment:
It won't compile due to these constants when `_WIN32_WINNT < _WIN32_WINNT_WIN8` since they're undefined (at least mingw is affected by this issue).

Reply:
Sorry I have no idea.. Here is the doc on msdn: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/ns-processthreadsapi-memory_priority_information
Maybe you need to include the header? or mingw shipped faulty header?

Reply:
Or I suppose we should define those constants (MEMORY_PRIORITY_NORMAL, etc.) in the above `#if` block ourselves?

Reply:
>Or I suppose we should define those constants (MEMORY_PRIORITY_NORMAL, etc.) in the above #if block ourselves?

Sure. I don't understand why you defined MEMORY_PRIORITY_INFORMATION but didn't define these constants which originally defined under the same conditions in windows headers. Or maybe MS compiler has different defines?

Reply:
>Or maybe MS compiler has different defines?

Both are using `DEFINES += _WIN32_WINNT=0x0601` in winconf.pri.
IIRC mingw headers isn't exactly the same as MSVC headers.

Are you able to submit a PR? as I don't have mingw env installed and can't verify the fix.

Reply:
>Or maybe MS compiler has different defines?

I found the macros is defined in winnt.h and is defined independent of windows version if I read correct (i.e. universally).